mitiq/calibration/settings.py

```diff
@@ -0,0 +1,137 @@
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from dataclasses import dataclass
+from functools import partial
+from itertools import product
+from typing import Any, Callable
+
+import networkx as nx
+
+from mitiq._typing import QPROGRAM
+from mitiq.benchmarks import (
+    generate_ghz_circuit,
+    generate_mirror_circuit,
+    generate_quantum_volume_circuit,
+    generate_rb_circuits,
+)
+from mitiq.pec import execute_with_pec
+from mitiq.zne import execute_with_zne
+from mitiq.zne.inference import ExpFactory, LinearFactory, RichardsonFactory
+from mitiq.zne.scaling import (
+    fold_gates_at_random,
+    fold_global,
+    insert_id_layers,
+)
+
+
+@dataclass
+class CircuitData:
+    circuit: QPROGRAM
+    dimensions: tuple[int, int]
+    type: str
+    two_qubit_gate_count: int
+
+
+class Settings:
+    """A class to store settings relating to error mitigation calibration."""
+
+    def __init__(
+        self,
+        mitigation_methods: list[str],
+        circuit_types: list[str],
+        circuit_dimensions: tuple[int, int],
+        method_params: dict[str, Any],
+    ):
+        self.mitigation_methods = mitigation_methods
+        self.method_params = method_params
+        self.circuit_types = circuit_types
+        self.circuit_dimensions = circuit_dimensions
+
+    def make_circuits(self) -> list[CircuitData]:
+        """Generate circuits specified by `circuit_types` and `circuit_dimensions`"""
+        circuits = []
+        nqubits, depth = self.circuit_dimensions
+        for circuit_type in self.circuit_types:
+            if circuit_type == "ghz":
+                circuit = generate_ghz_circuit(nqubits)
+            elif circuit_type == "rb":
+                circuit = generate_rb_circuits(nqubits, depth)[0]
+            elif circuit_type == "mirror":
+                circuit = generate_mirror_circuit(
+                    nlayers=depth,
+                    two_qubit_gate_prob=1.0,
+                    connectivity_graph=nx.complete_graph(nqubits),
+                )[0]
+            elif circuit_type == "qv":
+                circuit = generate_quantum_volume_circuit(nqubits, depth)[0]
+            else:
+                raise ValueError(
+                    f"invalid value passed for `circuit_types`. Must be one of `ghz`, `rb`, `mirror`, or `qv`, but got {circuit_type}"
+                )
+
+            circuit_dimensions = (len(circuit.all_qubits()), len(circuit))
+            two_qubit_gate_count = sum(
+                [len(op.qubits) > 1 for op in circuit.all_operations()]
+            )
+            circuits.append(
+                CircuitData(
+                    circuit,
+                    circuit_dimensions,
+                    circuit_type,
+                    two_qubit_gate_count,
+                )
+            )
+        return circuits
+
+    def mitigate_functions(self) -> list[tuple[str, Callable]]:
```

Comment:
This method could be changed into a `make_strategies()` method, returning `Strategy` objects instead of functions.